code/modules/awaymissions/gateway.dm

```diff
@@ -120,11 +120,14 @@ GLOBAL_DATUM(the_gateway, /obj/machinery/gateway/centerstation)
 		return
 	if(!powered())
 		return
+	var/targetdestination = stripped_input(user, "Enter the target destination ID", "Destination selection:", "MAIN_MISSION")
+	for(var/obj/machinery/gateway/centeraway/G in world)
```

Comment:
I don't know how, but this should be done better, isn't world every z level?

Reply:
I'll put the gateways into a global list in a bit and call from that.